libensemble/tests/unit_tests/test_persistent_aposmm.py

```diff
@@ -168,8 +168,88 @@ def test_standalone_persistent_aposmm_combined_func():
     assert persis_info.get("run_order"), "Standalone persistent_aposmm didn't do any localopt runs"
 
 
+@pytest.mark.extra
+def test_asktell_with_persistent_aposmm():
+    from math import gamma, pi, sqrt
+
+    import libensemble.gen_funcs
+    from libensemble.generators import APOSMM
+    from libensemble.message_numbers import FINISHED_PERSISTENT_GEN_TAG
+    from libensemble.sim_funcs.six_hump_camel import six_hump_camel_func
+    from libensemble.tests.regression_tests.support import six_hump_camel_minima as minima
+
+    libensemble.gen_funcs.rc.aposmm_optimizers = "nlopt"
+
+    n = 2
+    eval_max = 2000
+
+    gen_out = [("x", float, n), ("x_on_cube", float, n), ("sim_id", int), ("local_min", bool), ("local_pt", bool)]
+
+    gen_specs = {
+        "in": ["x", "f", "local_pt", "sim_id", "sim_ended", "x_on_cube", "local_min"],
+        "out": gen_out,
+        "user": {
+            "initial_sample_size": 100,
+            "sample_points": np.round(minima, 1),
+            "localopt_method": "LN_BOBYQA",
+            "rk_const": 0.5 * ((gamma(1 + (n / 2)) * 5) ** (1 / n)) / sqrt(pi),
+            "xtol_abs": 1e-6,
+            "ftol_abs": 1e-6,
+            "dist_to_bound_multiple": 0.5,
+            "max_active_runs": 6,
+            "lb": np.array([-3, -2]),
+            "ub": np.array([3, 2]),
+        },
+    }
+
+    my_APOSMM = APOSMM(gen_specs)
+    my_APOSMM.setup()
```

Comment:
Does this need to be separate to constructor?

Reply:
Perhaps you've seen already, but `setup()` sets attributes that can't be pickled, but need to be done anyway. So I believe a separate `generator.setup()` needs to exist.

But perhaps for outside-of-libE purposes, to save a line, this could resemble: `my_APOSMM = APOSMM(gen_specs, setup=True)`.